src/api/app/components/sign_up_component.rb

```diff
@@ -0,0 +1,18 @@
+# frozen_string_literal: true
+
+class SignUpComponent < ApplicationComponent
+  attr_accessor :submit_btn_text
+
+  def initialize(submit_btn_text:)
+    super
+    @submit_btn_text = submit_btn_text || 'Sign up'
+  end
+
+  def proxy_auth_enabled?
+    CONFIG['proxy_auth_mode'] == :on
```

Comment:
This view component would be easier to test if it doesn't depend on `CONFIG`. I would prefer to pass the config in the `initialize`.

Reply:
I'm passing it now, but IMO to stub some keys in a hash isn't really hard...